src/psyclone/domain/lfric/lfric_constants.py

```diff
@@ -145,8 +140,9 @@ def __init__(self):
         LFRicConstants.VALID_INTRINSIC_TYPES = supported_fortran_datatypes
 
         # Valid intrinsic types for field kernel argument data
-        # ('real' and 'integer').
-        LFRicConstants.VALID_FIELD_INTRINSIC_TYPES = ["real", "integer"]
+        # ('real', 'integer', and 'logical').
```

Comment:
I've just had at look at the latest LFRic trunk and, AFAICS, there's no support for logical fields?

Reply:
I wasn't sure either, but I checked with the tests, and there was an explicit test for logical fields:
```
@pytest.mark.parametrize(
...
     (lfric_psyir.LogicalFieldDataDataType,
      lfric_psyir.LogicalFieldDataDataSymbol,
      lfric_psyir.LfricLogicalScalarDataType,
      [Reference(
          lfric_psyir.NumberOfUniqueDofsDataSymbol(
              "ndofs", "w2",
              interface=ArgumentInterface(ArgumentInterface.Access.READ)))],
      {"fs": "w2"}),
...
def test_arrays(data_type, symbol, scalar_type, dims, attribute_map):
    '''Test the generated array datatypes and datasymbols are created
    correctly. This includes field datatypes and symbols which are
    kept as a separate list in psyir.py
```
and then there is `kernel_interface.py`:
```
    field_mapping = {
        "integer": lfric_psyir.IntegerFieldDataDataSymbol,
        "real": lfric_psyir.RealFieldDataDataSymbol,
        "logical": lfric_psyir.LogicalFieldDataDataSymbol}

```
I can't say if this is actually used anywhere, but that was my motivation to add that in. I am happy either way - I can also modify the test and remove logicals (which would make sense if it's not supported), but thought I'd better point out the two existing usages in PSylcone

Reply:
Oh, I think we can blame @rupertford for that :-) In that case, I suggest we leave it as you have it now.